cmd/internal/kube/injector/injector.go

```diff
@@ -0,0 +1,204 @@
+package injector
+
+import (
+	"bufio"
+	"bytes"
+	"github.com/pkg/errors"
+	"io"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kyamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/kube-openapi/pkg/util/sets"
+	"os"
+	"path/filepath"
+)
+
+type (
+	Injector interface {
```

Comment:
The interface name was a little confusing to me because this is a bit more like a wrapper around the data structure than like a visitor or actor.  It's more like a "Resources" or "Declarations".  But we can leave it as-is.

Reply:
Agreed! I initially intended for this interface to only have a single `Inject` method that took a set of resources and updated them, but it seems much better suited to be a wrapper and manipulator of Kubernetes resources.

I won't address this now, but if I have time later I'll refactor this interface's name into something more well-suited.